drv/sidecar-front-io/src/transceivers.rs

```diff
@@ -300,44 +301,148 @@ impl Transceivers {
         &self.fpgas[c as usize]
     }
 
-    // helper to write a value to CONTROL_PORTx registers given a mask
-    fn write_control_port<M: Into<FpgaPortMasks>>(
+    /// Executes a specified WriteOp (`op`) at `addr` for all ports specified by
+    /// the `mask`.
+    pub fn masked_port_op<M: Into<FpgaPortMasks>>(
```

Comment:
Nit: if this function is only ever called here, then it could be (a) not `pub` and (b) take `FpgaPortmasks` directly, leaving the generic conversions to the user-facing APIs.

Reply:
Good catch